analyzer/decrypt_crypto.cc

```diff
@@ -367,12 +405,16 @@ std::vector<uint8_t> HP_INFO_V2 = {0x00, 0x10, 0x0f, 0x74, 0x6c, 0x73, 0x31, 0x3
 
 class QuicPacketProtectionV2 : public QuicPacketProtection {
 public:
-    virtual std::vector<uint8_t>& GetInitialSalt() override { return INITIAL_SALT_V2; }
+    virtual bool Supports(uint32_t version) const override { return version == 0x6b3343cf; }
+
+    virtual std::vector<uint8_t>& GetInitialSalt(uint32_t version) const override {
+        assert(version == 0x6b3343cf);
```

Comment:
Making sure this assertion holds is some work, so I am wondering whether an `assert` isn't unnecessarily brittle. What do you think about e.g., throwing an exception instead so we don't forcibly tear down the calling process?

```suggestion
        if ( version != 0x6b3343cf )
            hilti::rt::internalError(hilti::rt::fmt("unexpected version '%d' for packet protection v2"));
```

Reply:
> Making sure this assertion holds is some work

You meant runtime work/overhead, or maintenance? There should be no overhead for production builds.

Anyway, I've removed the assertion completely, given the `Supports()` implementation and usage of this currently.

Reply:
I meant maintenance overhead. Removing it is even better.